### DISCUSSION TITLE
How to implement require()

### DISCUSSION THREAD

**Discussion by U-100** [sebastienros/jint#2007] [Q&A] [answered]
I basically did this:
`
engine.SetValue("require", new Action<string>(moduleName => engine.Modules.Import(moduleName)));
`
mymodule.js is one line:
`
export const name = "John";
`
now in my engine script:
`
require("./mymodule.js");
log(name);
`
It tells me that name is not defined. What am I doing wrong?

**Comment by Genteure** [comment]
Are you looking for something like this?

```js
import { name } from "./mymodule.js";
log(name);
```

https://github.com/sebastienros/jint?tab=readme-ov-file#using-modules
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#importing_features_into_your_script

**Reply by U-100** [reply]
Ok, that was the first thing I tried in my script actually. It gives:
`
Cannot use import statement outside a module (<anonymous>:2:1)
`

**Reply by U-100** [reply]
Maybe I should add a little more context. In C# I'm basically doing this:

```
string script = File.ReadAllText(scriptFilePath);
engine.Execute(script);
JsValue tickFunction = engine.GetValue("tick");
engine.Invoke(tickFunction);
```

and inside that script I need to import my module

**Comment by SebastianStehle** [comment]
Require returns an object with exported values. Probably something like this would work

```
const { name } = require(...);
log(name);
```

**Comment by U-100** [comment, marked as answer]
ok, I got around this problem by doing:
```
ObjectInstance module = engine.Modules.Import(team.scriptFile);
JsValue tickFunction = module.Get("tick");
engine.Invoke(tickFunction);
```
and now import { name } from "./mymodule.js"; works as expected in JavaScript
